timesketch/lib/datastores/opensearch.py

```diff
@@ -705,8 +728,13 @@ def search(
                 cause = ", ".join(error_items)
             else:
                 cause = str(e)
-
-            es_logger.error("Unable to run search query: %s", cause, exc_info=True)
+            log_message = (
+                f"Unable to run search query. Error: {cause}. "
+                f"Sketch ID: {sketch_id}. Indices: {indices}. "
+                f"Query String: '{query_string}'. Query Filter: {query_filter}. "
+                f"Query DSL: {query_dsl}"
+            )
+            os_logger.error(log_message, exc_info=True)
```

Comment:
For strings that are loaded into logger output we should always use %s lazy evaluation. This means the string formatting is deferred until it's absolutely necessary (i.e., when the log level is enabled and the message needs to be output). This ensures a better performance.

Reply:
ack